pennylane/devices/legacy_facade.py

```diff
@@ -0,0 +1,428 @@
+                return self._setup_adjoint_config(config)
+            if self._validate_device_method(tape):
+                config = replace(execution_config, gradient_method="device")
+                return self._setup_execution_config(config)
+
+        if execution_config.gradient_method == "backprop":
+            return self._setup_backprop_config(execution_config)
+        if execution_config.gradient_method == "adjoint":
+            return self._setup_adjoint_config(execution_config)
+        if execution_config.gradient_method == "device":
+            return self._setup_device_config(execution_config)
+
+        return execution_config
+
+    def supports_derivatives(self, execution_config=None, circuit=None) -> bool:
+        circuit = qml.tape.QuantumScript([], [], shots=self.shots) if circuit is None else circuit
+
+        if execution_config is None or execution_config.gradient_method == "best":
+            validation_methods = (
+                self._validate_backprop_method,
+                self._validate_adjoint_method,
+                self._validate_device_method,
+            )
+            return any(validate(circuit) for validate in validation_methods)
+
+        if execution_config.gradient_method == "backprop":
+            return self._validate_backprop_method(circuit)
+        if execution_config.gradient_method == "adjoint":
+            return self._validate_adjoint_method(circuit)
+        if execution_config.gradient_method == "device":
+            return self._validate_device_method(circuit)
+        return False
+
+    # pylint: disable=protected-access
+    def _create_temp_device(self, batch):
+        """Create a temporary device for use in a backprop execution."""
+        params = []
+        for t in batch:
+            params.extend(t.get_parameters(trainable_only=False))
+        interface = qml.math.get_interface(*params)
+        if interface == "numpy":
+            return self._device
+
+        mapped_interface = qml.workflow.execution.INTERFACE_MAP.get(interface, interface)
+
+        backprop_interface = self._device.capabilities().get("passthru_interface", None)
+        if mapped_interface == backprop_interface:
+            return self._device
+
+        backprop_devices = self._device.capabilities().get("passthru_devices", None)
+
+        if backprop_devices is None:
+            raise qml.DeviceError(f"Device {self} does not support backpropagation.")
+
+        if backprop_devices[mapped_interface] == self._device.short_name:
+            return self._device
+
+        # create new backprop device
+        expand_fn = self._device.expand_fn
+        batch_transform = self._device.batch_transform
+        if hasattr(self._device, "_debugger"):
+            debugger = self._device._debugger
+        else:
+            debugger = "No debugger"
+        tracker = self._device.tracker
+
+        backprop_devices = self._device.capabilities().get("passthru_devices", None)
```

Comment:
You already initialized `backprop_devices` above.

```suggestion
        tracker = self._device.tracker
```